scripts/zones/Aht_Urhgan_Whitegate/npcs/Kaduru-Haiduru.lua

```diff
@@ -0,0 +1,112 @@
+-----------------------------------
+-- Area: Aht Urhgan Whitegate
+-- NPC: Kaduru-Haiduru
+-- Teleport NPC
+-----------------------------------
+
+local function canUse_KaduruHaiduru_Service(player)
+    local caughtUsingShihuDanhuDate = player:getVar("Kaduru_ShihuDanhu_date")
+    local shihuDanhuEncounters = player:getVar("ShihuDanhu_Encounters")
+    local shihuDanhuDate = player:getVar("ShihuDanhu_TP_date")
+    local currentDate = os.date("%j")
+
+    if ((shihuDanhuEncounters < 1) or (shihuDanhuDate == 0) or
```

Comment:
individual things don't need wrapped in (stuff), only use to make order of operation clearer.

examples:

```lua
if (x and y) or (y and z) then
```

```lua
if x and y then
```

```lua
if x and y and z then
```

```lua
if x or y or z then
```

```lua
-- this is the bad example
if ((stuff) and (things)) then
```